src/commands/logs.ts

```diff
@@ -155,78 +49,188 @@ export default async (cmd: { json: boolean; open: boolean; setup: boolean; watch
     return open(url, { wait: false });
   }
 
-  /**
-   * Fetches the logs and prints the to the user.
-   * @param startDate {Date?} Get logs from this date to now.
-   */
-  async function fetchAndPrintLogs(startDate?: Date) {
-    spinner.setSpinnerTitle(`${oauthSettings.isLocalCreds ? LOG.LOCAL_CREDS : ''}${LOG.GRAB_LOGS}`).start();
-    // Create a time filter (timestamp >= "2016-11-29T23:00:00Z")
-    // https://cloud.google.com/logging/docs/view/advanced-filters#search-by-time
-    let filter = '';
-    if (startDate) {
-      filter = `timestamp >= "${startDate.toISOString()}"`;
-    }
-    // validate projectId
-    if (!projectId) {
-      return logError(null, ERROR.NO_GCLOUD_PROJECT);
-    }
-    if (!isValidProjectId(projectId)) {
-      return logError(null, ERROR.PROJECT_ID_INCORRECT(projectId));
-    }
-    try {
-      const logs = await logger.entries.list({
-        requestBody: {
-          resourceNames: [`projects/${projectId}`],
-          filter,
-          orderBy: 'timestamp desc',
-        },
-      });
-      // We have an API response. Now, check the API response status.
-      spinner.stop(true);
-      // Only print filter if provided.
-      if (filter.length) {
-        console.log(filter);
-      }
-      // Parse response and print logs or print error message.
-      // tslint:disable-next-line:no-any
-      const parseResponse = (response: any) => {
-        if (logs.status !== 200) {
-          switch (logs.status) {
-            case 401:
-              logError(
-                null,
-                oauthSettings.isLocalCreds ? ERROR.UNAUTHENTICATED_LOCAL : ERROR.UNAUTHENTICATED,
-              );
-            case 403:
-              logError(
-                null,
-                oauthSettings.isLocalCreds ? ERROR.PERMISSION_DENIED_LOCAL : ERROR.PERMISSION_DENIED,
-              );
-            default:
-              logError(null, `(${logs.status}) Error: ${logs.statusText}`);
-          }
-        } else {
-          printLogs(logs.data.entries);
-        }
-      };
-      parseResponse(logs);
-    } catch (error) {
-      spinner.stop(true);
-      logError(null, ERROR.PROJECT_ID_INCORRECT(projectId));
-    }
-  }
-
   // Otherwise, if not opening StackDriver, load StackDriver logs.
-  const oauthSettings = await loadAPICredentials();
   if (cmd.watch) {
     const POLL_INTERVAL = 6000; // 6s
     setInterval(() => {
       const startDate = new Date();
       startDate.setSeconds(startDate.getSeconds() - (10 * POLL_INTERVAL) / 1000);
-      fetchAndPrintLogs(startDate);
+      fetchAndPrintLogs(cmd.json, projectId, startDate);
     }, POLL_INTERVAL);
   } else {
-    fetchAndPrintLogs();
+    fetchAndPrintLogs(cmd.json, projectId);
+  }
+};
+
+/**
+ * Prints log entries
+ * @param entries {any[]} StackDriver log entries.
+ */
+export function printLogs(entries: logging_v2.Schema$LogEntry[] = [],
+                   formatJson: boolean) {
```

Comment:
Now additionally takes `formatJson` parameter (which corresponds to `cmd.json`) because it cannot access the `cmd` object from outside the main function.